### ISSUE TITLE
BSA tab

### ISSUE THREAD

**Issue by Utumno** [wrye-bash/wrye-bash#233]
- [ ] enable the BSA tab
- [ ] flesh it out -> extract, pack, list etc

**Comment by alt3rn1ty** [comment]
Utumno could the BSA Tab be hidden / disabled by default ?, possibly with a right click menu option somewhere to enable it ( similar to the current options on the Installers Tab Header Bar menu )

For reference a good topic to read on the impact of Mod Organiser including the ability to manipulate BSAs ..
http://forum.step-project.com/topic/5033-ramifications-of-bsa-extraction-in-mod-organizer/

I think that often times heated topic ( many many heated topics around the bazaars ) would stand as an example as to why I think Wrye Bash ought to hide its BSA capabilities by default from the outset. So that advanced users would be able to seek it out and enable it, but generally it would not be enabled for most users, and cause mod authors the same round of complex problem solving headaches they have had with MO users having the ability to manipulate the contents of BSAs.
I think it will certainly be a welcome addition for advanced users.

**Comment by Arthmoor** [comment]
I agree. If this is intended to give people a Bash UI for handling BSA files then it ought to be disabled by default with directions in the readme on how to get at it if people want it.

Please PLEASE don't go the same route as MO though and allow a way to sabotage the load order with BSA files. It's hard enough dealing with the MO zealots who think that's a good idea.

**Comment by Utumno** [comment]
Thanks for feedback - I have no clue about those BSA issues - I just wanted

> to give people a Bash UI for handling BSA

Anyway - I will disable it for 306 and will enable it on 307 if libsa works - if not no BSA tab - I patched libloadorder but I do not want to patch any more dlls, thanks. All tabs in wrye Bash can be enabled/disabled - so I will disable it by default, or I can even require an ini otion to have it enabled

Thanks for links but I can't go through posts and posts - so what is the real problem ? I mean with "sabotage the load order with BSA files"

**Comment by Arthmoor** [comment]
Basically MO has a means by which you can change the order BSAs load in. This goes entirely counter to how Skyrim expects this to happen, which is the game expects a BSA with the same name as the ESP to load in the same order the ESM/ESP files are listed. This has led to countless troubleshooting issues with people putting things in the wrong order and then trying to blame mod authors for it being messed up when it was the tool that allowed it to get messed up. Such features shouldn't just be "disabled by default" - they shouldn't exist at all.

**Comment by Utumno** [comment]
Ok - I was thinking (very vaguely) of allowing operations like extract, pack and list BSAs (or other stuff along these lines). Would this be ok (if I ever get round to it and if lbbsa.dll works) ?

**Comment by Arthmoor** [comment]
That would be fine, so long as there's never any weird functionality to alter the game's load order system. Which I don't think can even be done without some fort of virtual file system, but still.

**Comment by alt3rn1ty** [comment]
[quote]
Thanks for links but I can't go through posts and posts
[/quote]
Well all you need to read from the topic is the first post, z edited it multiple times to reflect all pertinent information brought up in the topic and other topics on the STEP forum.

And Post number 13 was Tannin's response to the concerns.

**Comment by Infernio** [comment]
I implemented extraction of FO4's texture BA2s on `inf-233-texture-ba2s`. Needs `inf-338-lz4` merged first, because I didn't want to have to deal with two divergent branches that heavily alter `bsa_files.py` 😛

**Comment by Infernio** [comment]
Totally forgot that I wanted to merge `inf-338-lz4` and `inf-233-texture-ba2s` together, so don't me merging that one real quick :eyes:
